avm/res/network/application-gateway/main.bicep

```diff
@@ -0,0 +1,622 @@
+metadata name = 'Network Application Gateways'
+metadata description = 'This module deploys a Network Application Gateway.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Application Gateway.')
+@maxLength(80)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Authentication certificates of the application gateway resource.')
+param authenticationCertificates array = []
+
+@description('Optional. Upper bound on number of Application Gateway capacity.')
+param autoscaleMaxCapacity int = -1
```

Comment:
If the `-1` is just intended as a 'user didn't set anything' value, it might be better to just make it nullable instead and check for that. The same for the next parameter `autoscaleMinCapacity`. Just a thought

Reply:
@AlexanderSehr updated: https://github.com/Azure/bicep-registry-modules/pull/835/commits/3bed15d676f8101a91cd32070d755d2c8842798b

Reply:
Realized I made a mistake in the above commit and fixed it: https://github.com/Azure/bicep-registry-modules/pull/835/commits/4b25b037c205c7e292be6c5047f6885c834595f4

Reply:
These changes are now resulting in the following error: 
```
Application Gateway /subscriptions/***/resourceGroups/dep-***-network.applicationgateway-nagmin-rg/providers/Microsoft.Network/applicationGateways/***nagmin001 must specify either Capacity or AutoscaleConfiguration for the selected SKU tier WAF_v2 (Code: ApplicationGatewayV2SkuMustSpecifyEitherCapacityOrAutoscaleConfiguration)
```
I don't know why `capacity: autoscaleMaxCapacity == null && autoscaleMinCapacity == null ? null : capacity` is not setting setting `capacity` in the default e2e test. We do not provide values for `autoscaleMaxCapacity` nor `autoscaleMinCapacity` in default test and `capacity` is set to 2. 

I am going to revert these changes. We can look at this later after I finish migrating the module.

Reply:
Will also take a look @ilhaan , thanks for testing 💪

Reply:
Ah I think the condition was just flipped.

It's used like this `autoscaleMaxCapacity > 0` which, if the parameter is nullable should translate to `autoscaleMaxCapacity != null` instead of `autoscaleMaxCapacity == null`, right?

Reply:
And sorry for already commenting on the PR. I realized only half way through that it's still in draft 😄